spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerResultAdapter.java

```diff
@@ -26,7 +26,7 @@
  * annotated method is operating in declarative mode.
  * @author Marius Bogoevici
  */
-public interface StreamListenerResultAdapter<R, B> {
+public interface StreamListenerResultAdapter<R, B, D> {
```

Comment:
Chatting with @sobychacko about this offline. It seems like we could skip the generic parameter and simply return a `Closeable` from `adapt`. This would give more flexibility to the core to act on the result (e.g. right now `StreamListenerResultAdapter` can't really do anything with the return value, because it doesn't know what it is. 

@artembilan - any thoughts?